kobo_backup.py

```diff
@@ -59,40 +63,68 @@ def main(args):
         )
         sys.exit()
 
     # Get the folder of the previous backup that occured
     try:
-        previous_backup = max(
-            glob.glob(os.path.join(backup_base_directory, "*/")), key=os.path.getmtime
-        )  
+        if args.compress:
+            previous_backup = max(
+                glob.glob(os.path.join(backup_base_directory, "*")),
+                key=os.path.getmtime,
+            )
+        else:
+            previous_backup = max(
+                glob.glob(os.path.join(backup_base_directory, "*/")),
+                key=os.path.getmtime,
+            )
     except ValueError:
         pass
 
     # Copy files
-    try: 
+    try:
         shutil.copytree(Path(kobo), backup_path)
     except OSError:  # some unrequired .Trashes will return 'operation not permitted'.
         pass
 
-    # Print size of last backup and current backup to stdout.
     try:
-        previous_backup
-        print(
-            f"The previous backup contained {sum(len(files) for _, _, files in os.walk(previous_backup))} files and was {get_size_format(get_directory_size(previous_backup))}."
-        )
-    except NameError:
-        pass
-
-    print(
-        f"Backup complete. Copied {sum(len(files) for _, _, files in os.walk(backup_path))} files with a size of {get_size_format(get_directory_size(backup_path))} to {backup_path}."
-    )
+        if args.compress:
+            compressed_backup_path = backup_path + ".tar.gz"
+            make_tarfile(compressed_backup_path, backup_path)
+        else:
+            pass
+    except Exception:
+        print("Failed to compress the backup")
+        sys.exit()
+    finally:
+        # Print size of last backup and current backup to stdout.
+        try:
+            previous_backup
+            print(
+                f"The previous backup contained {sum(len(files) for _, _, files in os.walk(previous_backup))} files and was {get_size_format(get_directory_size(previous_backup))}."
+            )
+        except NameError:
+            pass
+        if args.compress:
+            print(
+                f"Backup complete. Copied {sum(len(files) for _, _, files in os.walk(backup_path))} files with a total raw size of {get_size_format(get_directory_size(backup_path))} and a compressed size of {get_size_format(get_directory_size(compressed_backup_path))} to {compressed_backup_path}."
+            )
+            try:
+                shutil.rmtree(backup_path)
+            except Exception:
+                print("Failed to remove backup directory after compressing.")
+                sys.exit()
+            else:
+                pass
+        else:
+            print(
+                f"Backup complete. Copied {sum(len(files) for _, _, files in os.walk(backup_path))} files with a total size of {get_size_format(get_directory_size(backup_path))} to {backup_path}."
+            )
 
     # Only tested the below on Linux
     try:
```

Comment:
The `backup_path` variable in `subprocess.run(["xdg-open", backup_path])` (line 134) needs to be `compressed_backup_path` if user has chosen to compress.